api-gateway/graphql-api/modules/books/src/test/java/io/knotx/example/books/GraphQLHandlerFactoryTest.java

```diff
@@ -0,0 +1,131 @@
+package io.knotx.example.books;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.knotx.junit5.util.HoconLoader;
+import io.knotx.server.api.context.ClientRequest;
+import io.knotx.server.api.context.RequestContext;
+import io.knotx.server.api.context.RequestEvent;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.buffer.impl.BufferImpl;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.impl.ParsableHeaderValuesContainer;
+import io.vertx.ext.web.impl.ParsableMIMEValue;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.web.RoutingContext;
+import java.net.URL;
+import java.util.Collections;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+class GraphQLHandlerFactoryTest {
+
+  @Test
+  void expectBook(Vertx vertx, VertxTestContext testContext) throws Throwable {
+    GraphQLHandlerFactory factory = new GraphQLHandlerFactory();
+    getConfig(conf -> {
+      // given
+      String query = "{\"query\": \"{"
```

Comment:
Consider loading these from file

Reply:
I checked in in Intellij, there is no support for graphql query syntax.